packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding_email_confirmation/blocs/onboarding_email_confirmed_bloc.dart

```diff
@@ -0,0 +1,80 @@
+{{> licence.dart }}
+
+import 'package:rx_bloc/rx_bloc.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../base/common_services/onboarding_service.dart';
+import '../../base/extensions/error_model_extensions.dart';
+import '../../base/models/errors/error_model.dart';
+import '../../lib_router/blocs/router_bloc.dart';
+import '../../lib_router/router.dart';
+
+part 'onboarding_email_confirmed_bloc.rxb.g.dart';
+
+/// A contract class containing all events of the OnboardingEmailConfirmedBloC.
+abstract class OnboardingEmailConfirmedBlocEvents {
+  /// Verify the user's email with the provided token
+  void verifyEmail();
+
+  /// Redirect back to the login page in case of error
+  void goToLogin();
+
+  /// Redirect to the phone entry page to continue onboarding
+  void goToPhonePage();
+}
+
+/// A contract class containing all states of the OnboardingEmailConfirmedBloC.
+abstract class OnboardingEmailConfirmedBlocStates {
+  /// The loading state
+  Stream<bool> get isLoading;
+
+  /// The error state
+  Stream<ErrorModel> get errors;
+
+  /// Returns true when the email is verified
+  ConnectableStream<bool> get data;
+
+  /// The routing state for navigating to login/phone page
+  ConnectableStream<void> get onRouting;
+}
+
+@RxBloc()
+class OnboardingEmailConfirmedBloc extends $OnboardingEmailConfirmedBloc {
+  OnboardingEmailConfirmedBloc(
+    this._verifyEmailToken,
+    this._onboardingService,
+    this._routerBloc,
+  ) {
+    onRouting.connect().addTo(_compositeSubscription);
+    data.connect().addTo(_compositeSubscription);
+  }
+
+  final String _verifyEmailToken;
+  final OnboardingService _onboardingService;
+  final RouterBlocType _routerBloc;
+
+  @override
+  ConnectableStream<bool> _mapToDataState() => _$verifyEmailEvent
+      .startWith(null)
+      .throttleTime(const Duration(milliseconds: 200))
+      .switchMap((value) => _onboardingService
+          .confirmEmail(token: _verifyEmailToken)
+          .asResultStream())
+      .setResultStateHandler(this)
+      .whereSuccess()
+      .map((event) => true)
```

Comment:
Avoid bool steams that have `true` only value. Either use Steam<void> or Stream<BusinessModel>